src/builtin/list.rs

```diff
@@ -248,16 +248,15 @@ fn index(mut args: CallArgs, parser: &mut Parser<'_>) -> SassResult<Value> {
     args.max_args(2)?;
     let list = parser.arg(&mut args, 0, "list")?.as_list();
     let value = parser.arg(&mut args, 1, "value")?;
-    // TODO: find a way around this unwrap.
-    // It should be impossible to hit as the arg is
-    // evaluated prior to checking equality, but
-    // it is still dirty.
+    // TODO: find a way to propagate any errors here
     // Potential input to fuzz: index(1px 1in 1cm, 96px + 1rem)
     let index = match list.into_iter().position(|v| {
-        v.equals(value.clone(), args.span())
-            .unwrap()
-            .is_true(args.span())
-            .unwrap()
+        ValueVisitor::new(parser, args.span())
+            .equal(
+                HigherIntermediateValue::Literal(v),
+                HigherIntermediateValue::Literal(value.clone()),
```

Comment:
Do we need a clone here? Can't it compare to a reference?

Reply:
Almost certainly we could remove this clone (and in fact the reliance on the `ValueVisitor` altogether). I avoided doing so in this pull request specifically because of potential code duplication between multiple `equal` implementations and the fact that avoiding this single clone would be very much a microoptimization (that is, a 1 second improvement for a builtin list function would likely have the same effect as a 10ms improvement for selector parsing due to how common each are.)

I did make a note of this, but it is a very low priority (I might fix it when I work on simplifying value parsing in the distant future).